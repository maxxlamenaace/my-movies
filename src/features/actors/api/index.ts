import { getActorDetails } from './get-actor-details/get-actor-details';
import { getActorCombinedCredits } from './get-actor-combined-credits/get-actor-combined-credits';

export { getActorDetails, getActorCombinedCredits };
