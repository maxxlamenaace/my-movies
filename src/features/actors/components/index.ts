import ActorsGrid from './actors-grid/ActorsGrid';
import ActorItem from './actor-item/ActorItem';
import ActorInfos from './actor-infos/ActorInfos';

export { ActorsGrid, ActorItem, ActorInfos };
