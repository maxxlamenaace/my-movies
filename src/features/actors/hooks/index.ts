import { useActorDetails } from './use-actor-details/use-actor-details';

export { useActorDetails };
