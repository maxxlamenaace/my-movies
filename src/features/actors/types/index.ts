import { Media } from '@/features/medias';

export type Actor = {
  id: number;
  birthday: string;
  known_for_department: string;
  deathday?: string;
  name: string;
  also_known_as: string[];
  gender: number;
  biography: string;
  popularity: number;
  place_of_birth: string;
  profile_path: string;
  adult: boolean;
  imdb_id: string;
  homepage: string;
};

export type ActorImage = {
  id: number;
  profiles: ActorImageProfile[];
};

export type ActorImageProfile = {
  aspect_ratio: number;
  file_path: string;
  height: number;
  width: number;
};

export type ActorDetails = {
  credits: Media[];
} & Actor;
