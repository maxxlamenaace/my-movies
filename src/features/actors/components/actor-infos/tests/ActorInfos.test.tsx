import React from 'react';

import { render, cleanup } from '@testing-library/react';

import { ActorInfos } from '@/features/actors';
import { actor } from '@/tests/factories';

describe('ActorInfos', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders correctly', () => {
    const { asFragment } = render(<ActorInfos actor={actor()} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
