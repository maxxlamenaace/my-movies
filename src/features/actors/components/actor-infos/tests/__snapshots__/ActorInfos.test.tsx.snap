// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorInfos renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-vtpeil"
  >
    <div
      class="MuiBox-root css-1n04f71"
    >
      <div
        class="MuiBox-root css-pwp60y"
      />
    </div>
    <div
      class="MuiBox-root css-ye9s23"
    >
      <div
        class="css-1wi43tc-MuiStack-root"
      >
        <div
          class="css-1xq7zmv-MuiStack-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-mrmr51-MuiTypography-root"
          >
            name
          </h4>
          <div
            class="css-m69qwo-MuiStack-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-tjvwag-MuiTypography-root"
            >
              birthday
            </h6>
          </div>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-u2bbna-MuiTypography-root"
        >
          biography
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
